sort/sort.go

```diff
@@ -25,7 +25,13 @@ func MultithreadedSortComparators(comparators Comparators) Comparators {
 	copy(toBeSorted, comparators)
 
 	var wg sync.WaitGroup
-	chunks := chunk(toBeSorted, int64(runtime.NumCPU()))
+
+	numCPU := int64(runtime.NumCPU())
+	if numCPU%2 == 1 { // single core machine
+		numCPU++
+	}
```

Comment:
I don't understand why this works. Can you explain?

Either way, it would be nice to wrap our NumCPU() calls in some utility function that would do this for us.